aten/src/ATen/test/ivalue_test.cpp

```diff
@@ -308,6 +309,18 @@ TEST(IValueTest, FutureExceptions) {
   ASSERT_EQ(f3->tryRetrieveErrorMessage(), std::string("My Error"));
 }
 
+TEST(IValueTest, FutureSetError) {
+  auto f1 = c10::make_intrusive<ivalue::Future>(IntType::get());
+  f1->setError(std::make_exception_ptr(std::runtime_error("foo")));
+  try {
+    f1->setError(std::make_exception_ptr(std::runtime_error("bar")));
+    FAIL() << "Expected to throw";
+  } catch (std::exception& e) {
+    EXPECT_THAT(e.what(), ::testing::HasSubstr("Error already set"));
```

Comment:
nit: also test that it has the original error message? 
EXPECT_THAT(e.what(), ::testing::HasSubstr("foo"));